src/crimson/osd/pg.cc

```diff
@@ -975,9 +984,9 @@ PG::do_osd_ops_execute(
   FailureFunc&& failure_func)
 {
   assert(ox);
-  auto rollbacker = ox->create_rollbacker([this] (auto& obc) {
-    return obc_loader.reload_obc(obc).handle_error_interruptible(
-      load_obc_ertr::assert_all{"can't live with object state messed up"});
+  auto rollbacker = ox->create_rollbacker(
+    [object_context=duplicate_obc(obc)] (auto& obc) {
+    obc = duplicate_obc(object_context);
```

Comment:
Why `obc` needs to be duplicated twice (988 and 989)?  Is `duplicate_obc() unnecessary at 989 ?

Reply:
The first copy happens before `OpsExecuter::execute_op()` and serves as a backup of the original obc, while the second during the rollback and recovers the original obc.

Reply:
What can be wrong if the obc is only copied at the `do_osd_ops_execute()` time ?

Reply:
> What can be wrong if the obc is only copied at the `do_osd_ops_execute()` time ?

Sorry, I'm not following, could you elaborate about this a bit?

Reply:
> https://github.com/ceph/ceph/pull/59543#discussion_r1756169173

What can go wrong if:
```c++
auto rollbacker = ox->create_rollbacker(
    [object_context=duplicate_obc(obc)] (auto& obc) {
    obc = object_context;
});
```
?

Reply:
Ah:-D, yeah, changing

Reply:
@Matan-B I think the problem is actually here, when rolling back, we were not updating the one in the obc_registry but `OpsExecuter::obc`, will correct it.